### PR TITLE
support backfill id filtering for backfills

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Adds a `backfill_ids` filter to `BulkActionsFilter` so that the `backfill id` filter option on the runs feed can operate on backfills too.

I originally considered just adding a special case when fetching the runs feed where if the backfill id filter was passed we directly fetched the backfill. however, that required manually checking that all of the other filters matched the backfill, and that the cursor was respected, which felt prone to bugs. You can see the draft here if interested https://github.com/dagster-io/dagster/pull/24793

corresponding internal pr https://github.com/dagster-io/internal/pull/11818

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by jamiedemaria** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24799?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24799** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24799?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24745** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24745?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24755** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24755?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24754** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24754?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/execution/backfill.py [hunk 1]
```diff
@@ -62,13 +62,15 @@ class BulkActionsFilter:
...
         job_name (Optional[str]): Name of the job to query for. If blank, all job_names will be accepted.
+        backfill_ids (Optional[Sequence[str]]): A list of backfill_ids to filter by. If blank, all backfill_ids will be included
     """
 
     statuses: Optional[Sequence[BulkActionStatus]] = None
     created_before: Optional[datetime] = None
     created_after: Optional[datetime] = None
     tags: Optional[Mapping[str, Union[str, Sequence[str]]]] = None
     job_name: Optional[str] = None
+    backfill_ids: Optional[Sequence[str]] = None
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/execution/backfill.py:73]: BulkActionsFilter isn't part of the public api, but i could still mark the new param as experimental. of the ones i've added recently, this one seems the most likely to be removed/changed

**Review by prha — DISMISSED** [review]
can we split apart the core changes and the graphql changes?

do we not need gating for the internal storage tests which presumably run the run storage test suite?

**Comment by jamiedemaria** [comment]
> do we not need gating for the internal storage tests which presumably run the run storage test suite?

i implemented the internal support for the filter here https://github.com/dagster-io/internal/pull/11818, so is the idea that the gating would help in the case that we reverted the internal changes?

**Comment by prha** [comment]
well, wondering if we needed to support the behavior of the filter in the ursula storage.

File: python_modules/dagster/dagster_tests/storage_tests/utils/run_storage.py [hunk 2]
```diff
@@ -1730,6 +1730,38 @@ def test_backfill_simple_job_name_filtering(self, storage: RunStorage):
...
+        backfill = PartitionBackfill(
+            "backfill_1",
+            partition_set_origin=origin,
+            status=BulkActionStatus.REQUESTED,
+            partition_names=["a", "b", "c"],
+            from_failure=False,
+            tags={},
+            backfill_timestamp=time.time(),
+        )
+        storage.add_backfill(backfill)
```
Comment (OwenKephart) [inline python_modules/dagster/dagster_tests/storage_tests/utils/run_storage.py:1747]: nitpick, but could we also add another backfill with a different id just to make sure this filter is actually applied?

**Review by OwenKephart — APPROVED** [review]
note comment

File: python_modules/dagster/dagster/_core/execution/backfill.py (hunk 1 above)
Comment (OwenKephart) [reply python_modules/dagster/dagster/_core/execution/backfill.py:73]: it's not public, so I think we're fine here

**Comment by OwenKephart** [comment]
> well, wondering if we needed to support the behavior of the filter in the ursula storage.

whoops approved before seeing that, I assume this will fail in the internal test suite for that one?

**Pull request merged**